### DISCUSSION TITLE
Azure ADO repo not found

### DISCUSSION THREAD

**Discussion by nilekhc** [renovatebot/renovate#15816] [Request Help] [answered]
I have added `PROJECT/REPO` in the config.json under `repositories`. This is different project than where Renovate repo is created. But when pipeline is ran gives error `ERROR: Could not find repo in repo list (repository=PROJECT/REPO)`. It seems that it just trying to list repo from project where renovate is, and then erroring out.

Any idea why config.json is not honoring repos from different project.

Note: both Renovate project and Project mentioned is config.json are under same ORG.

**Comment by rarkins** [comment]
Please show your full config here

**Reply by nilekhc** [reply]
@rarkins Thanks. Here is the config and the log.

```js
module.exports = {
    logLevel: 'debug',
    platform: 'azure',
    endpoint: 'https://dev.azure.com/<ORG>/',
    token: process.env.TOKEN,
    hostRules: [
      {
        hostType: 'npm',
        matchHost: 'pkgs.dev.azure.com',
        username: 'apikey',
        password: process.env.TOKEN,
      },
    ],
    repositories: [
      'PROJ_1/REPO_1'
    ],
  };
```

This config is in the repo `PROJ_2/RENOVATE_REPO`. When pipeline is ran even though repo mentioned in `repositories` is `PROJ_1/REPO_1`, it only lists repo from PROJ_2/RENOVATE_REPO and errors with `repo not found` for `PROJ_1/REPO_1`. Both PROJ_1 and PROJ_2 are in the same ORG.

Here is the pipeline log:
```shell
found 0 vulnerabilities
 INFO: Repository started (repository=PROJ_1/REPO_1)
       "renovateVersion": "32.71.0"
ERROR: Could not find repo in repo list (repository=PROJ_1/REPO_1)
       "repos": [
         {
           "id": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
           "name": "RENOVATE_REPO",
           "url": "https://dev.azure.com/ORG/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/_apis/git/repositories/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
           "project": {
             "id": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
             "name": "PROJ_2",
             "url": "https://dev.azure.com/ORG/_apis/projects/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
             "state": "wellFormed",
             "revision": 203,
             "visibility": 1,
             "lastUpdateTime": "2022-05-31T21:41:04.520Z"
           },
           "defaultBranch": "refs/heads/main",
           "size": 728,
           "remoteUrl": "https://**redacted**@dev.azure.com/ORG/PROJ_2/_git/RENOVATE_REPO"
           "sshUrl": "git@ssh.dev.azure.com:v3/ORG/PROJ_2/RENOVATE_REPO",
           "webUrl": "https://dev.azure.com/ORG/PROJ_2/_git/RENOVATE_REPO"
           "isDisabled": false
         }
       ],
       "repo": null
ERROR: Repository is not found (repository=PROJ_1/REPO_1)
```

cc @JamieMagee

**Reply by JamieMagee** [reply, marked as answer]
I worked with @nilekhc offline to figure this out.

There are two specific settings that need to be disabled at both the organization level AND the project level. They only need to be disabled in the project which you are running Renovate in. For example, if you have a pipeline in `Project1` that runs Renovate and targets the repository `Project2/Repository1`, you only need to disable this setting in `Project1`.

The first is "Limit job authorization scope to current project for non-release pipelines". This setting restrict the access token available in the Azure Pipeline to only access resources in the same project. If you have this setting enabled, the token in the pipeline is for a service account called `YOUR-PROJECT-NAME Build Service (YOUR-ORGANIZATION-NAME)`, but if it's disabled the token is for a service account called `Project Collection Build Service (YOUR-ORGANIZATION-NAME)`.

The second is "Protect access to repositories in YAML pipelines". This setting restricts the access token available in the Azure Pipeline to prevent any access to repositories. This needs to be disabled to allow Renovate to clone the repo.

Here's a screenshot of what these options look like.

Organization level:
![image](https://user-images.githubusercontent.com/1358764/171497813-82f252c8-cb8b-43eb-a5c9-a3f4b966307e.png)

Project level:
![image](https://user-images.githubusercontent.com/1358764/171497960-61908b6d-e12b-44e7-9ef2-00b570f34ca1.png)

Both of these settings can be found under "Settings" (Organization or Project) -> "Pipelines" -> "Settings".

Once both of these options are disabled at the organization and project level you can follow our existing documentation about permissions[^1]

There were two key pieces of Azure DevOps documentation to help us understand the issue. Firstly [this page][2] on job authorization scopes, and secondly [this page][3] on managing build service account permissions

[2]: https://docs.microsoft.com/en-us/azure/devops/pipelines/process/access-tokens?view=azure-devops&tabs=yaml#job-authorization-scope
[3]: https://docs.microsoft.com/en-us/azure/devops/pipelines/process/access-tokens?view=azure-devops&tabs=yaml#manage-build-service-account-permissions

[^1]: https://docs.renovatebot.com/setup-azure-devops/#using-a-single-pipeline-to-update-multiple-repositories

**Reply by nilekhc** [reply]
Thank you so much @JamieMagee for debugging this with me.

**Reply by rwijngaa** [reply]
Updated link: https://docs.renovatebot.com/modules/platform/azure/#using-a-single-pipeline-to-update-multiple-repositories
